### PR TITLE
Replace deprecated Exodus API calls with non-deprecated versions

### PR DESCRIPTION
**Pull request by jwpeterson**
I split this up quite a bit so that if anything fails, it will hopefully be easy to bisect back to the breaking commit. If everything passes I could be convinced to squash the whole thing down into a single commit.

### CODE REVIEW

**Review by roystgnr — APPROVED** [review]
I like it better split up, personally.

**Comment by jwpeterson** [comment]
I ran this command from a test repository on latest devel just to see what would happen:
```
git merge --no-ff roystgnr/exodus_faces_soln jwpeterson/fix_exodus_deprecated
```
and there was a single, relatively trivial looking, merge conflict in exodusII_io_helper.C. If you want I can wait until #3233 is merged before attempting to merge this.

**Comment by roystgnr** [comment]
Nah, let's merge this whenever you're ready.  It's much more straightforward and I don't want to hold it up; #3233 is passing tests now but I'm not yet sure it's *working* working.

**Pull request merged**
